File: clients/rn/src/engine/persistence/InMemorySoulManifestStore.ts
/**
 * In-memory SoulManifestStore — no platform deps, useful for tests and mobile default.
 */

import type { ClientSoulManifest } from '../soul/SoulManifest';
import type { SoulManifestStore } from './SoulManifestStore';

export class InMemorySoulManifestStore implements SoulManifestStore {
  private store = new Map<string, ClientSoulManifest>();

  async save(manifest: ClientSoulManifest): Promise<void> {
    this.store.set(manifest.did, { ...manifest });
  }

  async load(did: string): Promise<ClientSoulManifest | null> {
    const m = this.store.get(did);
    return m ? { ...m } : null;
  }

  async delete(did: string): Promise<void> {
    this.store.delete(did);
  }

  async listDids(): Promise<string[]> {
    return [...this.store.keys()];
  }

  clear(): void {
    this.store.clear();
  }
}

File: clients/kmp/shared/src/commonMain/kotlin/org/wyrdsekai/app/engine/persistence/InMemorySoulManifestStore.kt
package org.wyrdsekai.app.engine.persistence

import org.wyrdsekai.app.engine.soul.ClientSoulManifest

/**
 * In-memory implementation of SoulManifestStore.
 * Suitable for testing and short-lived sessions. Not persistent across restarts.
 */
class InMemorySoulManifestStore : SoulManifestStore {
    private val store = mutableMapOf<String, ClientSoulManifest>()

    override suspend fun save(manifest: ClientSoulManifest) {
        store[manifest.did] = manifest
    }

    override suspend fun load(did: String): ClientSoulManifest? = store[did]

    override suspend fun delete(did: String) {
        store.remove(did)
    }

    override suspend fun listDids(): List<String> = store.keys.toList()
}
